CMakeLists.txt

```
@@ -389,6 +393,83 @@ endif()
 
 # ##############################################################################
 
+# If cuDSS is enabled and environment variables are set, install the cuDSS version
+if(USE_CUDSS AND DEFINED ENV{CUDA_PATH} AND DEFINED ENV{CUDSS_PATH})
+
+  # Force 32-bit integers for cuDSS compatibility
+  if(DLONG)
+    message(FATAL_ERROR "cuDSS requires 32-bit integers. Set DLONG=OFF or do not use -DDLONG=ON")
+  endif()
+
+  message(STATUS "CUDA_PATH set to $ENV{CUDA_PATH}")
+  message(STATUS "CUDSS_PATH set to $ENV{CUDSS_PATH}")
+  message(STATUS "Will install SCS-cuDSS (libscscudss).")
+
+  # Enable CUDA language
+  enable_language(CUDA)
+
+  set(CUDSSSRC ${LINSYS}/cudss/direct)
+
+  # Here we compile the direct cuDSS library
+  set(${PROJECT_NAME}_CUDSS ${PROJECT_NAME}cudss)
+  add_library(
+    ${${PROJECT_NAME}_CUDSS}
+    ${${PROJECT_NAME}_HDR} ${${PROJECT_NAME}_SRC} ${CUDSSSRC}/private.c
+    ${CUDSSSRC}/private.h)
+
+  target_include_directories(
+    ${${PROJECT_NAME}_CUDSS}
+    PRIVATE
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/${CUDSSSRC};${CMAKE_CURRENT_SOURCE_DIR}/${LINSYS}>"
+      "$ENV{CUDA_PATH}/include"
+      "$ENV{CUDSS_PATH}/include"
+  )
+
+  target_include_directories(
+    ${${PROJECT_NAME}_CUDSS}
+    PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+           "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/scs>")
+
+  target_compile_definitions(${${PROJECT_NAME}_CUDSS} PRIVATE ${COMPILER_OPTS})
+
+  target_link_directories(${${PROJECT_NAME}_CUDSS} PRIVATE
+    "$ENV{CUDA_PATH}/lib64"
+    "$ENV{CUDSS_PATH}/lib")
+
+  target_link_libraries(
+    ${${PROJECT_NAME}_CUDSS}
+    PRIVATE m
+            cudart
+            cusparse
+            cudss)
```

I does not seem to me that cusparse is used anywhere, right? Furthermore, the cmake imported target for cudart is called `CUDA::cudart`, see https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#cuda-runtime-library .

For what regards link to `m`, that is a platform specific thing, so it make sense to only link it if we are not on Windows.

```suggestion
  target_link_libraries(
    ${${PROJECT_NAME}_CUDSS}
    PRIVATE CUDA::cudart
            cudss
            $<$<NOT:$<PLATFORM_ID:Windows>>:m>)
```